hub/core/chunk_engine/generator.py

```diff
@@ -0,0 +1,106 @@
+import numpy as np
+from typing import Generator, Tuple
+
+from hub.core.chunk_engine.exceptions import ChunkGeneratorError
+
+
+def chunk(
+    content_bytes: bytes,
+    chunk_size: int,
+    last_chunk_num_bytes: int = None,
+) -> Generator[Tuple[bytes, int], None, None]:
+    """
+    Generator function that chunks bytes.
+
+    Chunking is the process of taking the input `content_bytes` & breaking it up into a sequence of smaller bytes called "chunks".
+    The sizes of each chunk are <= `chunk_size`.
+
+    Example 1 (Perfect Fit):
```

Comment:
These are tests in disguise. See below.

If these notions of "perfect fit" and "partial end" are not essential to the caller, they shouldn't be explained so laboriously.